tests/Unit/EmogrifierTest.php

```diff
@@ -737,6 +737,204 @@ public function emogrifyNotAppliesCssToNonMatchingElements($css, $expectedHtml)
+            '2 IDs & type' => '#p4 span#text',
+            '2 IDs & 2 types' => 'p#p4 span#text',
+            '2 IDs & class' => '.p-4#p4 #text',
+            '2 IDs & class & type' => '.p-4#p4 span#text',
+            '2 IDs & class & 2 types' => 'p.p-4#p4 span#text',
+            '2 IDs & 2 classes' => '.p-4.p-4#p4 #text',
+            '2 IDs & 2 classes & type' => '.p-4.p-4#p4 span#text',
+            '2 IDs & 2 classes & 2 types' => 'p.p-4.p-4#p4 span#text',
+        ];
+
+        $datasets = [];
+        $previousSelector = '';
+        $previousDescription = '';
+        foreach ($selectors as $description => $selector) {
+            if ($previousSelector !== '') {
+                $datasets[$description . ' more specific than ' . $previousDescription] = [
+                    '<span id="text"',
+                    $previousSelector,
+                    $selector
+                ];
+            }
+            $previousSelector = $selector;
+            $previousDescription = $description;
+        }
+
+        // broken: class more specific than 99 types (requires support for chaining `:not(h1):not(h1)...`)
+        $datasets['ID more specific than 99 classes'] = [
+            '<p class="p-4" id="p4"',
+            str_repeat('.p-4', 99),
+            '#p4'
+        ];
+
+        return $datasets;
+    }
+
+    /**
+     * @test
+     * @param string $matchedTagPart Tag expected to be matched by both selectors, without the closing '>',
+     *                               e.g. '<p class="p-1"'
+     * @param string $lessSpecificSelector A selector expression
+     * @param string $moreSpecificSelector Some other, more specific selector expression
+     * @dataProvider differentCssSelectorSpecificityDataProvider
+     */
+    public function emogrifyAppliesMoreSpecificCssSelectorToMatchingElements(
+        $matchedTagPart,
+        $lessSpecificSelector,
+        $moreSpecificSelector
+    ) {
+        $this->subject->setHtml(static::COMMON_TEST_HTML);
+        $this->subject->setCss(
+            $lessSpecificSelector . ' { color: red; } ' .
+            $moreSpecificSelector . ' { color: green; } ' .
+            $moreSpecificSelector . ' { background-color: green; } ' .
+            $lessSpecificSelector . ' { background-color: red; }'
+        );
+
+        $result = $this->subject->emogrify();
+
+        static::assertContains($matchedTagPart . ' style="color: green; background-color: green;"', $result);
+    }
+
+    /**
+     * @test Confirms the dataset is valid for the test, and the less specific selector does in fact match
+     * @param string $matchedTagPart Tag expected to be matched by selector, without the closing '>',
+     *                               e.g. '<p class="p-1"'
+     * @param string $lessSpecificSelector A selector expression
+     * @dataProvider differentCssSelectorSpecificityDataProvider
+     */
+    public function emogrifyWouldApplyLessSpecificCssSelectorToMatchingElements(
```

Comment:
The way I read this test is that is doesn't really check for less/more specific selectors, but tests that the given selectors work (without regard to specificity), i.e., we can remove this test as we already have tests for the individual selectors matching. Or am I missing anything here?

Reply:
Yes, this was just to confirm that the test datasets did in fact represent valid tests, and the lower precedence selector provided would in fact match the intended target if used alone.  However, since the test data is now written such that the lower precedence selector will be the same as the higher precedence one from the previous dataset, this is now unnecessary apart from the first dataset - but that uses the universal selector which obviously matches - and the final separate dataset with 99 classes - though for just a single dataset this could be confirmed 'by inspection'.

So I have removed this.